File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/components/SettingsFragmentComponent.java
package se.gustavkarlsson.skylight.android.dagger.components;

import dagger.Component;
import se.gustavkarlsson.skylight.android.background.UpdateScheduler;
import se.gustavkarlsson.skylight.android.dagger.scopes.FragmentScope;

@Component(dependencies = {
		ApplicationComponent.class
})
@SuppressWarnings("WeakerAccess")
@FragmentScope
public interface SettingsFragmentComponent {
	UpdateScheduler getUpdateScheduler();
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/components/SettingsFragmentComponent.kt
package se.gustavkarlsson.skylight.android.dagger.components

import dagger.Subcomponent
import se.gustavkarlsson.skylight.android.dagger.scopes.FragmentScope
import se.gustavkarlsson.skylight.android.gui.activities.settings.SettingsFragment

@Subcomponent
@FragmentScope
interface SettingsFragmentComponent {
	fun inject(settingsFragment: SettingsFragment)
}
